### PR TITLE
docs: add documentation for methods used to mount a UFS path to an Alluxio path

### PR DESCRIPTION
**Pull request by abnerferreira0212**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: 1542

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [ ] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [ ] Created tests for every new flow and method on this Pull Request
- [ ] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [ ] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [ ] Corrected all the problems and code smells identified by the quality tools
- [ ] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [ ] All unitary and integration tests ran by the CI system are passing
- [ ] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [ ] Sent this PR for a peer-review
- [ ] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java [hunk 1]
```diff
@@ -128,7 +128,23 @@ public Builder withSkipInitialSleep() {
     }
 
     /**
-     * @return the built retry mechanism
+     * Builds and returns a new exponential time-bounded retry mechanism.
+     * <p>
+     * Creates a new object of type {@link ExponentialTimeBoundedRetry} and
+     * returns it using these members variables:
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java:134]:
```suggestion
     * returns it using these member variables:
```

File: core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java [hunk 2]
```diff
@@ -62,7 +62,17 @@ public void resetMetaMasterConfigClient(MasterClientContext context) {
...
 
   /**
-   * @return empty if there is no exception during reinitialization, otherwise, return the exception
+   * Returns an exception if there is one during reinitialization.
+   * <p>
+   * Checks whether {@link #mException} is true or false. A truthy
+   * value indicates an exception occurred during reinitialization,
+   * in which case the exception is returned.
+   * <p>
+   * A falsy value indicates no exception took place, in which case
```
Comment (joaopedro-simbiose) [inline core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java:71]: "A false value indicates that no exception took place, then returns empty."

File: core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java [hunk 3]
```diff
@@ -62,7 +62,17 @@ public void resetMetaMasterConfigClient(MasterClientContext context) {
   }
 
   /**
-   * @return empty if there is no exception during reinitialization, otherwise, return the exception
+   * Returns an exception if there is one during reinitialization.
+   * <p>
+   * Checks whether {@link #mException} is true or false. A truthy
+   * value indicates an exception occurred during reinitialization,
+   * in which case the exception is returned.
```
Comment (joaopedro-simbiose) [inline core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java:69]: "A true value indicates that an exception occurred during reinitialization, so this exception is returned."

File: core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java [hunk 4]
```diff
@@ -62,7 +62,17 @@ public void resetMetaMasterConfigClient(MasterClientContext context) {
...
+   * <p>
+   * A falsy value indicates no exception took place, in which case
+   * the return is empty.
+   *
+   * @return  empty if there is no exception during reinitialization;
+   *          otherwise, the exception
```
Comment (joaopedro-simbiose) [inline core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java:75]: "...the occured exception"

File: core/common/src/main/java/alluxio/security/authentication/AuthType.java [hunk 5]
```diff
@@ -46,6 +46,11 @@
   ;
 
   /**
+   * Gets the string value of the authentication type
```
Comment (joaopedro-simbiose) [inline core/common/src/main/java/alluxio/security/authentication/AuthType.java:49]: Forgot to end the sentence with a dot (.)

File: core/common/src/main/java/alluxio/security/authentication/plain/SaslClientHandlerPlain.java [hunk 6]
```diff
@@ -81,6 +81,18 @@ public SaslClientHandlerPlain(String username, String password, String impersona
...
+   * Creates and returns a SASL client.
+   * <p>
+   * Instantiates a new object of type {@link SaslClient} using the provided
+   * username, password and impersonation user.
+   *
+   * @param   username          the username for the {@link PlainSaslClientCallbackHandler}
+   * @param   password          the password for the PlainSaslClientCallbackHandler
+   * @param   impersonationUser the impersonation user
+   * @return  the new SASL client with the provided information
+   * @throws UnauthenticatedException If a {@link SaslException} is thrown.
```
Comment (joaopedro-simbiose) [inline core/common/src/main/java/alluxio/security/authentication/plain/SaslClientHandlerPlain.java:94]: Use lowecase on "if"

File: core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java [hunk 7]
```diff
@@ -128,7 +128,23 @@ public Builder withSkipInitialSleep() {
...
+     *          - {@link ExponentialTimeBoundedRetry.Builder#mTimeCtx}, which
+     *          allows for time management;
```
Comment (joaopedro-simbiose) [inline core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java:136]: "which allows time management"

Comment (abnerferreira0212) [reply core/common/src/main/java/alluxio/retry/ExponentialTimeBoundedRetry.java]: Fixed.

File: core/common/src/main/java/alluxio/security/authentication/plain/SaslClientHandlerPlain.java (hunk 6 above)
Comment (abnerferreira0212) [reply core/common/src/main/java/alluxio/security/authentication/plain/SaslClientHandlerPlain.java]: Fixed,

File: core/common/src/main/java/alluxio/security/authentication/AuthType.java (hunk 5 above)
Comment (abnerferreira0212) [reply core/common/src/main/java/alluxio/security/authentication/AuthType.java]: Fixed.

File: core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java (hunk 4 above)
Comment (abnerferreira0212) [reply core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java]: Fixed.

File: core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java (hunk 3 above)
Comment (abnerferreira0212) [reply core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java:69]: Fixed.

File: core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java (hunk 2 above)
Comment (abnerferreira0212) [reply core/client/fs/src/main/java/alluxio/client/file/ConfigHashSync.java]: Fixed.
